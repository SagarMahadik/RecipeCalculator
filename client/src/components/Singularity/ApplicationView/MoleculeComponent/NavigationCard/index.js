import React from 'react';
import { CenterAlignedColumnContainer } from 'styles/Singularity/Style1.0/ContainerStyles/index.js';
import { NavigationButtonText } from 'styles/Singularity/Style1.0/TextStyles';
import { NavigationButton } from 'styles/Singularity/Style1.0/ButtonStyles';
const NavigationCard = ({ backgroundcolor, component, route }) => {
  return (
    <NavigationButton to={`/${route}`} backgroundcolor={backgroundcolor}>
      <NavigationButtonText>{component}</NavigationButtonText>
    </NavigationButton>
  );
};

export default NavigationCard;
