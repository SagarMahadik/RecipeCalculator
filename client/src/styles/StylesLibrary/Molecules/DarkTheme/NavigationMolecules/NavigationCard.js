import React from 'react';

import {
  CenterAlignedColumnContainerWithShadowBackground,
  CenterAlignedColumnContainer
} from 'styles/StylesLibrary/ElementalStyles/PositionStyles/index.js';

import {
  DashboardNavigationCard,
  DashboardNavigationCardText,
  DashboardNavigationCardContainer,
  DashboardNavigationCardBorder
} from 'styles/StylesLibrary/Atoms/NavigationCardElements/NavigationCards/index.js';

const NavigationCard = ({ module }) => {
  return (
    <CenterAlignedColumnContainer>
      <DashboardNavigationCardContainer>
        <DashboardNavigationCardBorder>
          <DashboardNavigationCard moduleName={module}>
            <DashboardNavigationCardText moduleName={module}>
              Raw Material
            </DashboardNavigationCardText>
          </DashboardNavigationCard>
        </DashboardNavigationCardBorder>
      </DashboardNavigationCardContainer>
    </CenterAlignedColumnContainer>
  );
};

export default NavigationCard;
