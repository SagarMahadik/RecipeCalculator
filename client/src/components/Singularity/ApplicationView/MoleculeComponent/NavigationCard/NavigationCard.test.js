import React from 'react';
import { render, cleanup } from '@testing-library/react';
import NavigationCard from 'components/Singularity/ApplicationView/MoleculeComponent/NavigationCard';
import { BrowserRouter as Router } from 'react-router-dom';
afterEach(cleanup);

it('Should match a snapshot', () => {
  const { asFragment } = render(
    <Router>
      <NavigationCard />
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});
