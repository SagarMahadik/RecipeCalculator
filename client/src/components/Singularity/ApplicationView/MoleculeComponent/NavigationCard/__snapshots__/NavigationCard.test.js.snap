// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match a snapshot 1`] = `
<DocumentFragment>
  <div
    class="ContainerStyles__ColumnContainer-rntyy-0 ContainerStyles__CenterAlignedColumnContainer-rntyy-2 jjFUUG"
  >
    <a
      class="sc-EHOje jdvhpy"
      href="/undefined"
    >
      <text
        class="TextStyles__PTSansText-sc-151f9kh-0 TextStyles__NavigationButtonText-sc-151f9kh-11 fdQvsx"
      />
    </a>
  </div>
</DocumentFragment>
`;
